CriticalMass/AnnotationController.swift

```diff
@@ -8,7 +8,7 @@
 
 import MapKit
 
-class AnnotationController<T: IdentifiableAnnnotation, K: MKAnnotationView> {
+class AnnotationController<T: IdentifiableAnnnotation<J>, K: AnnotationView, J> {
```

Comment:
I feel like the Generics are just just solving the friends use case but maybe I do not see the bigger picture. I could of course check other annotations too but might not be needed for them

Reply:
The Generics can be used for any AnnotationController that needs to provide an Object to their annotation views in a type safe way. That could also be used for events

Reply:
On a second look I see how this is supposed to work.
Though I can not see the type safety in this `BikeAnnotationController: AnnotationController<BikeAnnotation, BikeAnnoationView, Any>` and I do not see this working in an Array like in `MapViewController`

What I like is that there is one function to update annotations that I can override for my needs 👍 

What I do not like about is that: 
* `T` needs to be an IdentifiableAnnotation since that is MKPointAnnotation which is a concrete implementation  and not the protocol `MKAnnotation` 
* Makes this class hard to read and wrap heads around with a lot of wrapping and generic Types
* Uses `Location` type and not `CLLocation` which is preferred IMO (But I guess that is due to the LocationManager sending out Location types)
* What I meant in the beginning: It feels like the class was designed to fulfil the needs of a specific AnnotationType. So maybe it would make sense to have an empty `updateAnnotations` func since subclassing of the controller is necessary anyways

Reply:
Oh I totally forgot about the array 🙈 Trying to find a solution for that

Reply:
Sorry! you were totally right with your scepticism with using Generics here. Didn't see all the issues coming 🤦‍♂️ Not sure if it's  still worth it moving forward with this PR.. 

I removed the Array and replaced it with some properties witch introduced some code duplication in MapViewController. The code with all it generic Types is not nice to read. Not sure if the benefits of separation of concerns are strong enough to keep it